Source/WebCore/rendering/BreakLines.h

```diff
@@ -190,36 +194,44 @@ inline size_t BreakLines::nextBreakablePosition(CachedLineBreakIteratorFactory&
         }
 
         // Non-ASCII rapid lookup.
-        if (words != WordBreakBehavior::AutoPhrase) {
+        if constexpr (words != WordBreakBehavior::AutoPhrase) {
             if (!before.type)
                 before.type = classify<shortcutRules, nonBreakingSpaceBehavior>(before);
             after.type = classify<shortcutRules, nonBreakingSpaceBehavior>(after);
             // Short-circuit the commonest cases: letter + letter.
             int pair = before.type | after.type;
-            if (pair == (kAL | kAL)) {
-                if (words == WordBreakBehavior::BreakAll)
+            if (pair == kAL) {
+                if constexpr (words == WordBreakBehavior::BreakAll)
                     return i;
                 continue;
             }
             if ((pair | kAL) == (kID | kAL)) {
-                if (words == WordBreakBehavior::KeepAll)
+                if constexpr (words == WordBreakBehavior::KeepAll)
                     continue;
                 return i;
             }
+            // SP+AL SP+QU AL+AL AL+QU QU+QU QU+AL (after's SP is already filtered out).
+            if ((pair | kSP | kQU) & (kSP | kAL | kQU)) {
+                if constexpr (words == WordBreakBehavior::BreakAll)
+                    return i;
```

Comment:
I think this needs to be

```suggestion
                if constexpr (words == WordBreakBehavior::BreakAll)
                    if (pair == kAL)
                        return i;
```

Reply:
OK, changed!